src/ride/track.c

```diff
@@ -4880,15 +4881,491 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+    //pop     ebx
+    //pop     eax
+    //mov     ebx, dword_F4413E
+
+    //test    park_flags, 800h
+    //jz      short locret_6CDE16
+    if (RCT2_GLOBAL(RCT2_ADDRESS_PARK_FLAGS, uint32) & PARK_FLAGS_NO_MONEY) {
+        return 0;
+    }
+
+    return RCT2_GLOBAL(0xF4413E, money32);
+}
+
+
+
+money32 loc_6CDE51() {
+	// mov     ebx, 80000000h
+	return MONEY32_UNDEFINED;
+}
+
+money32 loc_6CDE4C() {
+	// pop     edi
+	// pop     edx
+	// pop     ecx
+	// pop     ebx
+	// pop     eax
+	
+	return loc_6CDE51();
+}
+
+money32 loc_6CDE43() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = 0;
+	
+	return loc_6CDE4C();
+}
+
+money32 loc_6CDE38() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = 954; // empty string?
+	
+	// jmp     short loc_6CDE51
+	return loc_6CDE51();
+}
+
+money32 loc_6CDE2D() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+	
+	// jmp     short loc_6CDE51
+	return loc_6CDE51();
+}
+
+money32 loc_6CDE22() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_RIDE_CANT_BUILD_THIS_UNDERWATER;
+	
+	return loc_6CDE4C();
+}
+
+money32 loc_6CDE17() {
+	// mov     game_command_error_text, 0
+	RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, rct_string_id) = STR_CAN_ONLY_BUILD_THIS_ABOVE_GROUND;
+	
+	return loc_6CDE4C();
+}
+
+money32 loc_6CD9C5(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp) {
+    sint8 var_14 = -0x14;
+    sint8 var_10 = -0x10;
+    sint8 var_C = -0xC;
+    sint8 var_8 = -0x8;
+    sint8 var_4 = -0x4;
+
+    // pop     edx
+    // pop     ecx
+    // pop     eax
+
+    uint16 x = (uint16) (eax & 0xFFFF); // ax
+    uint8 flags = (uint8) (ebx & 0xFF); // bl
+    uint8 direction = (uint8) ((ebx & 0xFF00) >> 8); // bh
+    uint16 y = (uint16) (ecx & 0xFFFF); // cx
+    uint8 rideIndex = (uint8) (edx & 0xFF); // dl
+    uint8 mode = (uint8) ((ebx & 0xFF00) >> 8); // dh
+    uint16 z = (uint16) (edi & 0xFFFF); // di
+
+    // push    eax
+    // push    ebx
+    // push    ecx
+    // push    edx
+    // push    edi
+
+    // and     ax, 0FFE0h
+    uint16 temp_ax = (uint16) floor2(x, 32);
+
+    // and     cx, 0FFE0h
+    uint16 temp_cx = (uint16) floor2(y, 32);
+
+    // mov     bl, dl
+    uint8 bl = rideIndex;
+    // mov     dx, di
+    uint16 temp_dx = z;
+    // shr     dx, 3
+    temp_dx = temp_dx >> 3;
+
+    // movzx   esi, cx
+    uint32 temp_esi = y;
+
+    // shl     esi, 8
+    temp_esi = temp_esi << 8;
+
+    // or      si, ax
+    temp_esi = (temp_esi & 0xFFFF0000) + (temp_esi & temp_ax);
+
+    // shr     esi, 3
+    temp_esi = temp_esi >> 3;
+
+    // mov     esi, tile_map_element_pointers[esi]
+    rct_map_element *temp_esi_map = TILE_MAP_ELEMENT_POINTER(temp_esi);
+
+
+    while (true) {
```

Comment:
To loop through map elements we normally do a do while (!map_element_is_last_for_tile(mapElement++)); loop.